src/AWS.Deploy.Orchestration/DeploymentSettingsHandler.cs

```diff
@@ -91,5 +100,54 @@ public async Task ApplySettings(DeploymentSettings deploymentSettings, Recommend
 
             throw new InvalidDeploymentSettingsException(DeployToolErrorCode.DeploymentConfigurationNeedsAdjusting, errorMessage.Trim());
         }
+
+        public async Task SaveSettings(string filePath, Recommendation recommendation, CloudApplication cloudApplication, OrchestratorSession orchestratorSession)
+        {
+            var parentDirectory = Path.GetDirectoryName(filePath);
+            if (string.IsNullOrEmpty(parentDirectory))
+            {
+                var message = $"Failed to save deployment settings because {filePath} is not a valid path";
+                throw new FailedToSaveDeploymentSettingsException(DeployToolErrorCode.FailedToSaveDeploymentSettings, message);
+            }
+            if (!_directoryManager.Exists(parentDirectory))
+            {
+                var message = $"Failed to save deployment settings because {filePath} does not exist on disk";
+                throw new FailedToSaveDeploymentSettingsException(DeployToolErrorCode.FailedToSaveDeploymentSettings, message);
+            }
+
+            var deploymentSettings = new DeploymentSettings
+            {
+                AWSProfile = orchestratorSession.AWSProfileName,
+                AWSRegion = orchestratorSession.AWSRegion,
+                ApplicationName = cloudApplication.Name,
+                RecipeId = cloudApplication.RecipeId,
+                Settings = new Dictionary<string, object>()
+            };
+
+            var optionSettings = recommendation.GetConfigurableOptionSettingItems();
+            foreach (var optionSetting in optionSettings)
+            {
+                var id = optionSetting.FullyQualifiedId;
+                var value = _optionSettingHandler.GetOptionSettingValue(recommendation, optionSetting);
+                deploymentSettings.Settings[id] = value;
```

Comment:
We should talk with Irene if we should be writing all of the values or just the ones that are different then the default. Seems like we are creating a file that is overly complex looking but it does give you a full list of all of the possible values.

Reply:
I think we should persist all values given that deploying via a config file is a slightly advance use case and users would prefer to have complete transparency over all the deployment settings. Also, we do not publicly document the default values for any of our option settings which makes it difficult for users to obtain this info without peeking into the recipe source code.